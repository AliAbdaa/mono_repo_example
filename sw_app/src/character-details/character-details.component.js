// @flow
import React, {PureComponent} from 'react';
import * as styles from "./character-details.module.scss";

export type CharacterDetailsProps = {
  name: string,
  height: string,
  mass: string,
  hair_color: string,
  skin_color: string,
  eye_color: string,
  birth_year: string,
  gender: string,
  url: string,
}

export class CharacterDetails extends PureComponent<CharacterDetailsProps> {
  render() {
    return (
      <div  className={styles.tile}>
        {this.props.name}
      </div>
    )
  }
}