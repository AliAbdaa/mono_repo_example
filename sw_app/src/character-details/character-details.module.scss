.tile {
  margin: 10px;
  display: flex;
  background-color: #999;
  color: rgba(251, 221, 42, 1);
  flex-wrap: wrap;
  font-size: 20px;
}

.title {
  font-size: 50px;
  font-weight: 300;
  margin: 20px;
  justify-content: center;
}

.row {
  padding: 0 50px;
  justify-content: space-between;
  text-transform: capitalize;
}

.col {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.col100 {
  flex: 0 0 100%;
}

.col50 {
  flex: 0 0 50%;
}

.color {
  display: inline-flex;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  margin: 0 5px;
  border: 1px solid black;
}