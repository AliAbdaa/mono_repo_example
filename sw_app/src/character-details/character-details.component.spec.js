import React from 'react';
import {shallow} from 'enzyme';
import {CharacterDetails} from "./character-details.component";

describe("character-details", () => {
  it("renders correctly", () => {
    const component = shallow(<CharacterDetails
      name={"test-name"}
      url={"test-url.com"}
      height={"1337"}
      mass={"100"}
      eye_color="green"
      birth_year="1989"
    />);

    expect(component).toMatchSnapshot();
  });
});