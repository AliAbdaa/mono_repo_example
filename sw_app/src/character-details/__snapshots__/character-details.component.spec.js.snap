// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character-details renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterDetails
    birth_year="1989"
    eye_color="green"
    height="1337"
    mass="100"
    name="test-name"
    url="test-url.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=""
        >
          test-name
        </div>,
        <div
          className=""
        >
          <TranslatedMessage
            localeKey="height"
          />
          <span>
            1337
          </span>
        </div>,
        <div
          className=""
        >
          <TranslatedMessage
            localeKey="mass"
          />
          <span>
            100
          </span>
        </div>,
        <div
          className=""
        >
          <TranslatedMessage
            localeKey="eyeColor"
          />
          <span
            style={
              Object {
                "backgroundColor": "green",
              }
            }
          />
        </div>,
        <div
          className=""
        >
          <TranslatedMessage
            localeKey="birthDay"
          />
          <span>
            1989
          </span>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test-name",
          "className": "",
        },
        "ref": null,
        "rendered": "test-name",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TranslatedMessage
              localeKey="height"
            />,
            <span>
              1337
            </span>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "localeKey": "height",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1337",
            },
            "ref": null,
            "rendered": "1337",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TranslatedMessage
              localeKey="mass"
            />,
            <span>
              100
            </span>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "localeKey": "mass",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "100",
            },
            "ref": null,
            "rendered": "100",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TranslatedMessage
              localeKey="eyeColor"
            />,
            <span
              style={
                Object {
                  "backgroundColor": "green",
                }
              }
            />,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "localeKey": "eyeColor",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "style": Object {
                "backgroundColor": "green",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TranslatedMessage
              localeKey="birthDay"
            />,
            <span>
              1989
            </span>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "localeKey": "birthDay",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1989",
            },
            "ref": null,
            "rendered": "1989",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            test-name
          </div>,
          <div
            className=""
          >
            <TranslatedMessage
              localeKey="height"
            />
            <span>
              1337
            </span>
          </div>,
          <div
            className=""
          >
            <TranslatedMessage
              localeKey="mass"
            />
            <span>
              100
            </span>
          </div>,
          <div
            className=""
          >
            <TranslatedMessage
              localeKey="eyeColor"
            />
            <span
              style={
                Object {
                  "backgroundColor": "green",
                }
              }
            />
          </div>,
          <div
            className=""
          >
            <TranslatedMessage
              localeKey="birthDay"
            />
            <span>
              1989
            </span>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test-name",
            "className": "",
          },
          "ref": null,
          "rendered": "test-name",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TranslatedMessage
                localeKey="height"
              />,
              <span>
                1337
              </span>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "localeKey": "height",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1337",
              },
              "ref": null,
              "rendered": "1337",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TranslatedMessage
                localeKey="mass"
              />,
              <span>
                100
              </span>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "localeKey": "mass",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "100",
              },
              "ref": null,
              "rendered": "100",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TranslatedMessage
                localeKey="eyeColor"
              />,
              <span
                style={
                  Object {
                    "backgroundColor": "green",
                  }
                }
              />,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "localeKey": "eyeColor",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "style": Object {
                  "backgroundColor": "green",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TranslatedMessage
                localeKey="birthDay"
              />,
              <span>
                1989
              </span>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "localeKey": "birthDay",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1989",
              },
              "ref": null,
              "rendered": "1989",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
